### PR TITLE
RUMM-1810 Add missing header in traced request

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

We are adding the missing header in the traced request: `x-datadog-sampled: 1` which indicates that this request is selected for sampling.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java [hunk 1]
```diff
@@ -49,6 +50,9 @@ public void inject(final DDSpanContext context, final TextMapInject carrier) {
       for (final Map.Entry<String, String> entry : context.baggageItems()) {
         carrier.put(OT_BAGGAGE_PREFIX + entry.getKey(), HttpCodec.encode(entry.getValue()));
       }
+
+      // makes this request trace selected for sampling
+      carrier.put(SELECTED_FOR_SAMPLE_KEY, "1");
```
Comment (0xnm) [inline dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java:55]: is there a relationship between `x-datadog-sampling-priority` and `x-datadog-sampled`? If priority is say `0`, this should be still `1`?

Comment (mariusc83) [reply dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java:55]: Is what I want to ask also during the daily.

Comment (0xnm) [reply dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java:55]: I didn't find values other than `1` in the DD code https://github.com/search?p=4&q=org%3ADataDog+%22x-datadog-sampled%22&type=Code

Anyway, iOS SDK is using `1` https://github.com/DataDog/dd-sdk-ios/blob/887b1429b8ec3804b194783aa423f989fcaac06d/Sources/Datadog/Tracing/Propagation/TracingHTTPHeaders.swift#L27

But at the same time on iOS `x-datadog-sampling-priority` is always `1`, unlike on Android.

**Review by 0xnm — APPROVED** [review]

File: dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java:55]: Out of consistency with iOS and React Native, let's make sure the `x-datadog-sampling-priority` is also always one.

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java (hunk 1 above)
Comment (ncreated) [reply dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java]: You can read more on this in https://github.com/DataDog/dd-sdk-ios/pull/575, notably that these values were taken from [this doc](https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=browserrum#how-are-rum-resources-linked-to-traces).

**Review by 0xnm — APPROVED** [review]
LGTM!

Minor: on a formatting side there are many whitespace changes (indent, line breaks?), maybe we can revert them?

**Review by xgouchet — COMMENTED** [review]
Did you reformat this file?

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**